h2o-core/src/test/java/water/fvec/VecTest.java

```diff
@@ -148,4 +146,50 @@ public void map(Chunk c) {
     source.remove(new Futures()).blockForPending();
     con.remove(new Futures()).blockForPending();
   }
+
+  @Test public void testChunkForChunkIdxAfterVecUpdate() {
+    try {
+      Scope.enter();
+      Vec v1 = Scope.track(makeSeq(2 * FileVec.DFLT_CHUNK_SIZE, false));
+      Vec v2 = Scope.track(makeSeq(2 * FileVec.DFLT_CHUNK_SIZE, false));
+      Chunk c0 = v1.chunkForChunkIdx(0);
+      c0._vec = v2; // inject any vec into the cached POJO, this simulates a stale Vec reference
+      Chunk c0c = v1.chunkForChunkIdx(0);
+      assertSame(c0, c0c);
+      // the vec reference was updated in the existing live object
```

Comment:
@michalkurka so is it expected behaviour that we change ref in `c0._vec` from `v2` to `v1` by just calling `v1.chunkForChunkIdx`? Is it the change of semantics that you were worried about?

Reply:
@michalkurka I maybe understand the problem that you described in the description. If we mutate not on the home node and send MRTask to `home` the following condition will return true `v == this` and it will return `c` - a stale version. But how do we define `home`? isn't it just the node from where we call MRTask? I'm asking because I wonder how to run into the case that you have described....how to call MRTask not from `home`?

Reply:
Home nodes are determined by where the underlying key that references the Vec is homed.

Reply:
@michalkurka But I don't see any logic that is dependant on keys ( that could probably distinguish home from not_home case ). Do we have code like `isHomeKey(key): Boolean`?
 I can only see that we have `v == this` condition that is checking addresses of the objects and just for local copy we can say that `it was already filled in`.  I was trying to understand your concern expressed in the issue's description but still I'm not sure that I can say that I understand how we addressed this concern. Lack of knowledge on my side.